drivers/docker/driver.go

```diff
@@ -0,0 +1,1250 @@
+	}
+}
+
+func (d *Driver) RecoverTask(handle *drivers.TaskHandle) error {
+	if _, ok := d.tasks.Get(handle.Config.ID); ok {
+		return nil
+	}
+
+	var handleState taskHandleState
+	if err := handle.GetDriverState(&handleState); err != nil {
+		return fmt.Errorf("failed to decode driver task state: %v", err)
+	}
+
+	client, _, err := d.dockerClients()
+	if err != nil {
+		return fmt.Errorf("failed to get docker client: %v", err)
+	}
+
+	container, err := client.InspectContainer(handleState.ContainerID)
+	if err != nil {
+		return fmt.Errorf("failed to inspect container for id %q: %v", handleState.ContainerID, err)
+	}
+
+	reattach, err := shared.ReattachConfigToGoPlugin(handleState.ReattachConfig)
+	if err != nil {
+		return err
+	}
+
+	dlogger, dloggerPluginClient, err := docklog.ReattachDockerLogger(reattach)
+	if err != nil {
+		return fmt.Errorf("failed to reattach to docker logger process")
+	}
+
+	h := &taskHandle{
+		client:                client,
+		waitClient:            waitClient,
+		dlogger:               dlogger,
+		dloggerPluginClient:   dloggerPluginClient,
+		logger:                d.logger.With("container_id", container.ID),
+		task:                  handle.Config,
+		containerID:           container.ID,
+		containerImage:        container.Image,
+		doneCh:                make(chan bool),
+		waitCh:                make(chan struct{}),
+		removeContainerOnExit: d.config.GC.Container,
+		net:                   handleState.DriverNetwork,
+	}
+
+	d.tasks.Set(handle.Config.ID, h)
+	go h.collectStats()
+	go h.run()
+
+	return nil
+}
+
+func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *structs.DriverNetwork, error) {
+	if _, ok := d.tasks.Get(cfg.ID); ok {
+		return nil, nil, fmt.Errorf("taskConfig with ID '%s' already started", cfg.ID)
+	}
+
+	var driverConfig TaskConfig
+
+	if err := cfg.DecodeDriverConfig(&driverConfig); err != nil {
+		return nil, nil, fmt.Errorf("failed to decode driver config: %v", err)
+	}
+
+	handle := drivers.NewTaskHandle(pluginName)
+	handle.Config = cfg
+
+	// Initialize docker API clients
+	client, _, err := d.dockerClients()
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to connect to docker daemon: %s", err)
+	}
+
+	id, err := d.createImage(cfg, &driverConfig, client)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	containerCfg, err := d.createContainerConfig(cfg, &driverConfig, id)
+	if err != nil {
+		d.logger.Error("failed to create container configuration", "image_name", driverConfig.Image,
+			"image_id", id, "error", err)
+		return nil, nil, fmt.Errorf("Failed to create container configuration for image %q (%q): %v", driverConfig.Image, id, err)
+	}
+
+	startAttempts := 0
+CREATE:
+	container, err := d.createContainer(client, containerCfg, &driverConfig)
+	if err != nil {
+		d.logger.Error("failed to create container", "error", err)
+		return nil, nil, nstructs.NewRecoverableError(fmt.Errorf("failed to create container: %v", err), nstructs.IsRecoverable(err))
```

Comment:
There's a little [WrapRecoverable](https://github.com/hashicorp/nomad/blob/d7f02ed474424d8639d0a2333d3980bd94349a61/nomad/structs/structs.go#L8515-L8520) error to help with this pattern.
```suggestion
		return nil, nil, nstructs.WrapRecoverable(fmt.Sprintf("failed to create container: %v", err), err)
```